cpp/benchmarks/join/join_benchmark_common.hpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <benchmark/benchmark.h>
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/join.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+
+#include <fixture/benchmark_fixture.hpp>
+#include <synchronization/synchronization.hpp>
+
+#include <vector>
+
+#include "generate_input_tables.cuh"
+
+template <typename key_type,
+          typename payload_type,
+          bool Nullable,
+          bool is_conditional = false,
+          typename state_type,
+          typename Join>
+static void BM_join(state_type& state, Join JoinFunc)
+{
+  auto const build_table_size = [&]() {
+    if constexpr (std::is_same<state_type, benchmark::State>::value) {
+      return static_cast<cudf::size_type>(state.range(0));
+    }
+    if constexpr (std::is_same<state_type, nvbench::state>::value) {
+      return static_cast<cudf::size_type>(state.get_int64("Build Table Size"));
+    }
+  }();
+  auto const probe_table_size = [&]() {
+    if constexpr (std::is_same<state_type, benchmark::State>::value) {
+      return static_cast<cudf::size_type>(state.range(1));
+    }
+    if constexpr (std::is_same<state_type, nvbench::state>::value) {
+      return static_cast<cudf::size_type>(state.get_int64("Probe Table Size"));
+    }
+  }();
```

Comment:
default return missing. Is this not throwing a warning during compilation?

Reply:
The two `if constexpr`s are to get rid of the warning you mentioned where NVCC will warn about a missing return when using an if constexpr/else.